### PR TITLE
[WIP] [#2760 finish] additional expenses - finish issue ticket

### PR DESCRIPTION
**Pull request by DrewAPeterson7671**
### What github issue is this PR for, if any?
Finish remaining UI items for #2760

### What changed, and why?
1.  Style Other Expenses to match Wireframe
2.  Fix refresh that takes the user to the top of the page each time "Add Another Expense" is clicked.
3. Verify handling to ensure a description is required and user warning provided if an expense has no description.

### How will this affect user permissions?
- Volunteer permissions:  Shouldn't have an effect based on login
- Supervisor permissions:
- Admin permissions:

### Feelings gif (optional)
What gif best describes your feeling working on this issue? https://giphy.com/
How to embed:
`alt text`

### Feedback please? (optional)
We are very interested in your feedback! Please give us some :) https://forms.gle/1D5ACNgTs2u9gSdh9

### CODE REVIEW

**Comment by compwron** [comment]
Current issue missing this error
Screen Shot 2022-03-08 at 6 56 18 PM

**Comment by compwron** [comment]
current CI issue
```

rspec ./spec/system/all_casa_admins/casa_orgs/casa_admins/new_spec.rb:8 # all_casa_admins/casa_orgs/casa_admins/new validates and creates new admin
rspec ./spec/controllers/case_contacts_controller_spec.rb:91 # CaseContactsController POST #create with valid params creates and assigns @case_contact
rspec ./spec/controllers/case_contacts_controller_spec.rb:141 # CaseContactsController POST #create with valid params when a casa case is selected renders the casa case show template
rspec ./spec/controllers/case_contacts_controller_spec.rb:130 # CaseContactsController POST #create with valid params when a casa case is selected creates a new case contact for each selected case
rspec ./spec/controllers/case_contacts_controller_spec.rb:166 # CaseContactsController POST #create with valid params with miles driven sets miles driven
rspec ./spec/system/all_casa_admins/new_spec.rb:7 # all_casa_admins/new validates and creates new all casa admin
rspec ./spec/system/all_casa_admins/edit_spec.rb:53 # all_casa_admin_edit_spec with invalid parameters does not update password
rspec ./spec/system/hearing_types/new_spec.rb:14 # hearing_types/new errors with invalid name
rspec ./spec/models/contact_type_spec.rb:23 # ContactType#update can't update to an invalid name
rspec ./spec/system/casa_cases/new_spec.rb:67 # casa_cases/new when the case number field is not filled does not create a new case
rspec ./spec/system/casa_cases/new_spec.rb:80 # casa_cases/new when the case number already exists in the organization does not create a new case
rspec ./spec/models/volunteer_spec.rb:293 # Volunteer invitation expiration expires invitation token after one year
rspec ./spec/system/judges/new_spec.rb:13 # judges/new adds new judge
rspec ./spec/system/casa_admins/edit_spec.rb:29 # casa_admins/edit with invalid data shows error message for empty email
rspec ./spec/models/contact_type_group_spec.rb:5 # ContactTypeGroup does not have duplicate names
rspec ./spec/system/all_casa_admins/casa_orgs/new_spec.rb:13 # all_casa_admins/casa_orgs/new as an all casa admin lets admin create a casa org
rspec ./spec/system/all_casa_admins/casa_orgs/new_spec.rb:46 # all_casa_admins/casa_orgs/new as an all casa admin requires name
rspec ./spec/system/casa_admins/new_spec.rb:6 # casa_admins/new validates and creates new admin
rspec ./spec/models/mileage_rate_spec.rb:24 # MileageRate#effective_date is unique within is_active and casa_org
rspec ./spec/system/users/edit_spec.rb:33 # users/edit volunteer user displays password errors messages when user is unable to set a password
rspec ./spec/system/users/edit_spec.rb:121 # users/edit when admin is not able to update the profile without display name as an admin
rspec ./spec/system/users/edit_spec.rb:136 # users/edit when admin displays password errors messages when admin is unable to set a password
rspec ./spec/requests/casa_cases_spec.rb:162 # /casa_cases as an admin POST /create invalid request with invalid parameters also respond to json
rspec ./spec/requests/case_contacts_spec.rb:51 # /case_contacts logged in as volunteer POST /create with valid parameters does create two new CaseContacts
rspec ./spec/requests/case_contacts_spec.rb:70 # /case_contacts logged in as volunteer POST /create with valid parameters when the URL contains ?success=true redirects to casa_case#show
rspec ./spec/requests/casa_admins_spec.rb:91 # /casa_admins PUT /casa_admins/:id when logged in as admin, but invalid data also respond as json
rspec ./spec/requests/all_casa_admins/casa_orgs_spec.rb:75 # AllCasaAdmin::CasaOrgs POST /create when failure also responds as json
rspec ./spec/system/case_contacts/new_spec.rb:8 # case_contacts/new when admin does not display empty or hidden contact type groups; can create CaseContact
rspec ./spec/system/case_contacts/new_spec.rb:85 # case_contacts/new when admin should allow expanding or hiding if notes are more than 100 characters
rspec ./spec/system/case_contacts/new_spec.rb:50 # case_contacts/new when admin should display full text in table if notes are less than 100 characters
rspec ./spec/system/case_contacts/new_spec.rb:164 # case_contacts/new when admin with HTML in notes renders HTML correctly on the index page
rspec ./spec/system/case_contacts/new_spec.rb:357 # case_contacts/new volunteer user submits the form when no note was added
rspec ./spec/system/case_contacts/new_spec.rb:326 # case_contacts/new volunteer user delete local storage notes after successfuly submited
rspec ./spec/system/case_contacts/new_spec.rb:386 # case_contacts/new volunteer user submits the form when note is added and confirmed
rspec ./spec/system/case_contacts/new_spec.rb:447 # case_contacts/new volunteer user with invalid inputs re-renders the form with errors, but preserving all previously entered selections
rspec ./spec/system/contact_types/edit_spec.rb:15 # contact_types/edit errors with invalid name
rspec ./spec/models/supervisor_spec.rb:22 # Supervisor invitation expiration expires invitation token after two weeks
rspec ./spec/system/contact_types/new_spec.rb:15 # contact_types/new errors with invalid name
rspec ./spec/requests/all_casa_admins_spec.rb:182 # /all_casa_admins PATCH /update_password with invalid parameters also responds as json
rspec ./spec/requests/all_casa_admins_spec.rb:64 # /all_casa_admins POST /create with invalid parameters also responds as json
rspec ./spec/requests/all_casa_admins_spec.rb:106 # /all_casa_admins PATCH /update with invalid parameters also responds as json
rspec ./spec/requests/court_dates_spec.rb:281 # /casa_cases/:casa_case_id/court_dates/:id PATCH /update with invalid parameters renders a successful response displaying the edit template
rspec ./spec/requests/court_dates_spec.rb:241 # /casa_cases/:casa_case_id/court_dates/:id POST /create invalid request with invalid parameters renders a successful response (i.e. to display the 'new' template)
rspec ./spec/requests/casa_org_spec.rb:66 # CasaOrg as an admin PATCH /update with invalid parameters also responds as json
rspec ./spec/models/casa_admin_spec.rb:36 # CasaAdmin invitation expiration expires invitation token after two weeks
```

**Comment by compwron** [comment]
@DrewAPeterson7671 how are you feeling about this? :)

**Comment by DrewAPeterson7671** [comment]
Its good, I know what the issue is and roughly what to do.  I should get it finished tomorrow and hopefully it will pass testing.

**Comment by compwron** [comment]
Screen Shot 2022-03-22 at 7 16 16 PM

**Comment by compwron** [comment]
Failing in CI:
```
rspec ./spec/system/judges/new_spec.rb:13 # judges/new adds new judge
rspec ./spec/system/casa_admins/new_spec.rb:6 # casa_admins/new validates and creates new admin
rspec ./spec/system/all_casa_admins/new_spec.rb:7 # all_casa_admins/new validates and creates new all casa admin
rspec ./spec/requests/all_casa_admins/casa_orgs_spec.rb:75 # AllCasaAdmin::CasaOrgs POST /create when failure also responds as json
rspec ./spec/requests/casa_admins_spec.rb:91 # /casa_admins PUT /casa_admins/:id when logged in as admin, but invalid data also respond as json
rspec ./spec/system/contact_types/edit_spec.rb:15 # contact_types/edit errors with invalid name
rspec ./spec/requests/casa_cases_spec.rb:162 # /casa_cases as an admin POST /create invalid request with invalid parameters also respond to json
rspec ./spec/models/contact_type_group_spec.rb:5 # ContactTypeGroup does not have duplicate names
rspec ./spec/models/supervisor_spec.rb:22 # Supervisor invitation expiration expires invitation token after two weeks
rspec ./spec/requests/all_casa_admins_spec.rb:64 # /all_casa_admins POST /create with invalid parameters also responds as json
rspec ./spec/requests/all_casa_admins_spec.rb:106 # /all_casa_admins PATCH /update with invalid parameters also responds as json
rspec ./spec/requests/all_casa_admins_spec.rb:182 # /all_casa_admins PATCH /update_password with invalid parameters also responds as json
rspec ./spec/system/casa_cases/new_spec.rb:67 # casa_cases/new when the case number field is not filled does not create a new case
rspec ./spec/system/casa_cases/new_spec.rb:97 # casa_cases/new when the case number already exists in the organization does not create a new case
rspec ./spec/system/casa_cases/new_spec.rb:78 # casa_cases/new when the court date field is not filled does not create a new case
rspec ./spec/models/volunteer_spec.rb:293 # Volunteer invitation expiration expires invitation token after one year
rspec ./spec/system/contact_types/new_spec.rb:15 # contact_types/new errors with invalid name
rspec ./spec/models/mileage_rate_spec.rb:24 # MileageRate#effective_date is unique within is_active and casa_org
rspec ./spec/system/all_casa_admins/edit_spec.rb:53 # all_casa_admin_edit_spec with invalid parameters does not update password
rspec ./spec/system/hearing_types/new_spec.rb:14 # hearing_types/new errors with invalid name
rspec ./spec/models/contact_type_spec.rb:23 # ContactType#update can't update to an invalid name
rspec ./spec/system/users/edit_spec.rb:121 # users/edit when admin is not able to update the profile without display name as an admin
rspec ./spec/system/users/edit_spec.rb:136 # users/edit when admin displays password errors messages when admin is unable to set a password
rspec ./spec/system/users/edit_spec.rb:33 # users/edit volunteer user displays password errors messages when user is unable to set a password
rspec ./spec/requests/casa_org_spec.rb:66 # CasaOrg as an admin PATCH /update with invalid parameters also responds as json
rspec ./spec/requests/court_dates_spec.rb:281 # /casa_cases/:casa_case_id/court_dates/:id PATCH /update with invalid parameters renders a successful response displaying the edit template
rspec ./spec/requests/court_dates_spec.rb:241 # /casa_cases/:casa_case_id/court_dates/:id POST /create invalid request with invalid parameters renders a successful response (i.e. to display the 'new' template)
rspec ./spec/system/casa_admins/edit_spec.rb:29 # casa_admins/edit with invalid data shows error message for empty email
rspec ./spec/models/casa_admin_spec.rb:36 # CasaAdmin invitation expiration expires invitation token after two weeks
rspec ./spec/system/all_casa_admins/casa_orgs/casa_admins/new_spec.rb:8 # all_casa_admins/casa_orgs/casa_admins/new validates and creates new admin
rspec ./spec/system/all_casa_admins/casa_orgs/new_spec.rb:46 # all_casa_admins/casa_orgs/new as an all casa admin requires name
rspec ./spec/system/all_casa_admins/casa_orgs/new_spec.rb:13 # all_casa_admins/casa_orgs/new as an all casa admin lets admin create a casa org

```

File: config/locales/en.yml [hunk 1]
```diff
@@ -112,3 +112,11 @@ en:
     thank_you_6: One of the greatest gifts you can give is your time!
     thank_you_7: Those who can do, do. Those who can do more, volunteer.
     thank_you_8: Volunteers do not necessarily have the time, they just have the heart.
+  errors:
```
Comment (compwron) [inline config/locales/en.yml:115]: @DrewAPeterson7671 maybe this is where all the test issues are coming from?

Comment (DrewAPeterson7671) [reply config/locales/en.yml]: It absolutely was!  Only one code climate issue to deal with.  Everything else passed.

**Comment by DrewAPeterson7671** [comment]
This only remaining test that fails is codeclimate for the case_contacts_controller.rb for the method on line 150.  I was just looking at this and I agree the nested if/else structures add complexity.  However, this would be alleviated a bit when the FeatureFlag is eventually taken out.  What if I take out lines 153 to 159 into their own method?

**Comment by compwron** [comment]
This is looking good!! Let's merge it and then clean up codeclimate :)

**Review by compwron — APPROVED** [review]

**Pull request merged**